bitwarden_license/bit-common/src/tools/reports/risk-insights/services/critical-apps.service.ts

```diff
@@ -94,6 +94,27 @@ export class CriticalAppsService {
     this.orgId.next(orgId);
   }
 
+  // Drop a critical app for a given organization
+  // Only one record can be dropped at a time
+  async dropCriticalApp(orgId: OrganizationId, selectedUrl: string) {
+    const record = this.criticalAppsList.value
+      .filter((f) => f.organizationId === orgId && f.uri === selectedUrl)
+      .map((f) => ({
+        organizationId: f.organizationId,
+        passwordHealthReportApplicationIds: [f.id],
+      }));
+
+    if (record.length === 0) {
+      return;
+    }
+
+    // drop one record  - only one record can be dropped at a time
+    await this.criticalAppsApiService.dropCriticalApp(record[0]);
+
+    const updatedList = this.criticalAppsList.value.filter((f) => f.uri !== selectedUrl);
+    this.criticalAppsList.next(updatedList);
+  }
```

Comment:
🎨  This could be a little more concise and avoid unnecessary use of `map`. Also since we're referring to the entity elsewhere as `app` we may want to stick with that here for consistency as the name `record` could be confused with the [Record[(https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkeys-type) utility.

```suggestion
// Drop a critical app for a given organization
// Only one app may be dropped at a time
async dropCriticalApp(orgId: OrganizationId, selectedUrl: string) {
  const app = this.criticalAppsList.value.find(
    f => f.organizationId === orgId && f.uri === selectedUrl
  );

  if (!app) return;

  await this.criticalAppsApiService.dropCriticalApp({
    organizationId: app.organizationId,
    passwordHealthReportApplicationIds: [app.id],
  });

  this.criticalAppsList.next(
    this.criticalAppsList.value.filter(f => f.uri !== selectedUrl)
  );
}
```